### DISCUSSION TITLE
Add single() and singleOrDefault() for query builder

### DISCUSSION THREAD

**Discussion by theminer3746** [laravel/framework#40347] [Ideas] [answered]
Currently, Laravel has `first()` which returns the first result, `firstOrFail()` which returns the first result or throw an exception if no result was found, and `firstOrDefault()` which returns the first result or a default if no result was found.

I propose implementing `single()` which returns the first result if there is exactly one result and throw an exception if there are many or none. `singleOrDefault()` works in similiar way but returns a default value if no result was found and throw an exception of multiple results were found.

The method name and general idea was inspired by LINQ.

**Comment by koenhoeijmakers** [comment, marked as answer]
This already exists but its named `sole()`
